src/ray/object_manager/connection_pool.cc

```diff
@@ -85,6 +96,21 @@ void ConnectionPool::Remove(ReceiverMapType &conn_map, const ClientID &client_id
   connections.erase(connections.begin() + pos);
 }
 
+void ConnectionPool::Remove(SenderMapType &conn_map, const ClientID &client_id,
+                            const std::shared_ptr<SenderConnection> &conn) {
+  auto it = conn_map.find(client_id);
+  if (it == conn_map.end()) {
+    return;
+  }
+  auto &connections = it->second;
+  int64_t pos =
+      std::find(connections.begin(), connections.end(), conn) - connections.begin();
+  if (pos >= static_cast<int64_t>(connections.size())) {
+    return;
+  }
+  connections.erase(connections.begin() + pos);
```

Comment:
So under the hood here somehow the file descriptor is actually getting closed in some destructor?

Reply:
The file descriptor is closed once the `shared_ptr` goes out of scope.